src/utils/liquidityMining.ts

```diff
@@ -242,3 +243,131 @@ export function getStakedAmountUSD(campaign: CurrencyAmount, nativeCurrencyUSDPr
     parseUnits(campaign.multiply(nativeCurrencyUSDPrice).toFixed(USD.decimals), USD.decimals).toString()
   )
 }
+
+interface SubgraphToken {
+  address: string
+  symbol: string
+  name: string
+  decimals: string
+}
+
+export interface SubgraphPair {
+  address: string
+  reserve0: string
+  reserve1: string
+  reserveNativeCurrency: string
+  reserveUSD: string
+  totalSupply: string
+  token0: SubgraphToken
+  token1: SubgraphToken
+  liquidityMiningCampaigns: SubgraphLiquidityMiningCampaign[]
+}
+
+interface TokenAmountParams {
+  token: SubgraphToken
+  tokensInCurrentChain: { [address: string]: { token: Token } }
+  chainId: ChainId
+  reserve: string
+}
+
+export const getTokenAmount = ({ token, tokensInCurrentChain, chainId, reserve }: TokenAmountParams) => {
+  const tokenChecksummedAddress = getAddress(token.address)
+  const tokenA =
+    tokensInCurrentChain && tokensInCurrentChain[tokenChecksummedAddress]?.token
+      ? tokensInCurrentChain[tokenChecksummedAddress].token
+      : new Token(chainId, tokenChecksummedAddress, parseInt(token.decimals, 10), token.symbol, token.name)
+  return new TokenAmount(tokenA, parseUnits(reserve, token.decimals).toString())
+}
+
+interface PairWithLiquidityMiningCampaignParams {
+  rawPair: SubgraphPair
+  tokensInCurrentChain: { [address: string]: { token: Token } }
+  chainId: ChainId
+  nativeCurrency: Currency
+  kpiTokens: KpiToken[]
+}
+
+export const getPairWithLiquidityMiningCampaign = ({
+  rawPair,
+  tokensInCurrentChain,
+  chainId,
+  kpiTokens,
+  nativeCurrency,
+}: PairWithLiquidityMiningCampaignParams) => {
+  const { reserveNativeCurrency, totalSupply, token0, token1, reserve0, reserve1, liquidityMiningCampaigns } = rawPair
+
+  const tokenAmountA = getTokenAmount({ token: token0, tokensInCurrentChain, chainId, reserve: reserve0 })
+  const tokenAmountB = getTokenAmount({ token: token1, tokensInCurrentChain, chainId, reserve: reserve1 })
+
+  const pair = new Pair(tokenAmountA, tokenAmountB)
+
+  const campaigns = liquidityMiningCampaigns.map(campaign => {
+    return toLiquidityMiningCampaign(
+      chainId,
+      pair,
+      totalSupply,
+      reserveNativeCurrency,
+      kpiTokens,
+      campaign,
+      nativeCurrency
+    )
+  })
+  pair.liquidityMiningCampaigns = campaigns
+
+  return pair
+}
+
+export const getLowerTimeLimit = () =>
+  Math.floor(
+    DateTime.utc()
+      .minus(Duration.fromObject({ days: 150 }))
+      .toSeconds()
+  )
+
+export const getRewardTokenAddressFromPair = (pair: SubgraphPair) =>
+  pair.liquidityMiningCampaigns.flatMap(campaign => campaign.rewards.map(reward => reward.token.address.toLowerCase()))
+
+export const sortActiveCampaigns = (activeCampaigns: any) =>
+  activeCampaigns.sort((a: any, b: any) => {
```

Comment:
We can get better typing with interface like

```ts
// Name sucks, God bless you for finding better one
export interface DisplayedMiningCampaign {
      campaign: LiquidityMiningCampaign | SingleSidedLiquidityMiningCampaign
      staked: boolean
      containsKpiToken: boolean
    }

export const sortActiveCampaigns = (activeCampaigns: DisplayedMiningCampaign[]) =>
// inference rocks!  
   activeCampaigns.sort((a, b) => {
/// ...
```

With this type we can also improve `useAllLiquidityMiningCampaigns` which I pointed in other comment

Reply:
True! It was just to get going but forget to add it. Thanks. 

As a name,  `MiningCampaign`, should work?